test/e2e/node/node_problem_detector.go

```diff
@@ -380,7 +378,7 @@ func getNpdPodStat(ctx context.Context, f *framework.Framework, nodeName string)
 
 	hasNpdPod := false
 	for _, pod := range summary.Pods {
-		if !strings.HasPrefix(pod.PodRef.Name, "npd") {
+		if !strings.HasPrefix(pod.PodRef.Name, "node-problem-detector") {
```

Comment:
npd.yaml used a daemonset name that didn't match the values in the upstream manifests. https://github.com/kubernetes/node-problem-detector/blob/master/deployment/node-problem-detector.yaml

kops deploys npd's daemonset with the name `node-problem-detector` so it is expected all kubernetes clusters are bootstrapped using the correct manifests provided by the component/project maintainer

Reply:
/cc @dims or @justinsb